src/main/scala/mesosphere/marathon/api/v2/InfoResource.scala

```diff
@@ -18,32 +24,26 @@ class InfoResource @Inject() (
     schedulerService: MarathonSchedulerService,
     mesosLeaderInfo: MesosLeaderInfo,
     leaderInfo: LeaderInfo,
-    // format: OFF
-    protected val config: MarathonConf
-      with HttpConf with EventConfiguration with HttpEventConfiguration with LeaderProxyConf
+  // format: OFF
+  protected val config: MarathonConf
+    with HttpConf
+    with EventConfiguration
+    with HttpEventConfiguration
+    with LeaderProxyConf
+    with OfferMatcherManagerConfig
+    with OfferProcessorConfig
+    with DebugConf
+    with LaunchQueueConfig
+    with LaunchTokenConfig
 ) extends RestResource {
   // format: ON
 
   // Marathon configurations
   private[this] lazy val marathonConfigValues = Json.obj(
-    "master" -> config.mesosMaster.get,
-    "failover_timeout" -> config.mesosFailoverTimeout.get,
-    "framework_name" -> config.frameworkName.get,
-    "ha" -> config.highlyAvailable.get,
-    "checkpoint" -> config.checkpoint.get,
-    "local_port_min" -> config.localPortMin.get,
-    "local_port_max" -> config.localPortMax.get,
-    "executor" -> config.defaultExecutor.get,
-    "hostname" -> config.hostname.get,
-    "webui_url" -> config.webuiUrl.get,
-    "mesos_role" -> config.mesosRole.get,
-    "task_launch_timeout" -> config.taskLaunchTimeout.get,
-    "reconciliation_initial_delay" -> config.reconciliationInitialDelay.get,
-    "reconciliation_interval" -> config.reconciliationInterval.get,
-    "marathon_store_timeout" -> config.marathonStoreTimeout.get,
-    "mesos_user" -> config.mesosUser.get,
-    "leader_proxy_connection_timeout_ms" -> config.leaderProxyConnectionTimeout.get,
-    "leader_proxy_read_timeout_ms" -> config.leaderProxyReadTimeout.get)
+    ConfigToMap.convertToMap(config).map {
```

Comment:
`ConfigToMap.convertToMap` explicitly returns a `Map[String, Option[Any]]`.

This return type is basically unusable, so that should also be addressed.  However, to clear the type error you should be able to stringify the values using [Map.stringToMap](http://www.scala-lang.org/api/current/index.html#scala.collection.Map@mapValues[C](f:B=>C):scala.collection.Map[A,C]) as follows:

```scala
ConfigToMap.convertToMap(config).mapValues(_.map(_.toString)).map {
  // now values are of type Option[String],
  // and scalac can find a Writes[String]...
}
```